### PR TITLE
Show non writable folders during move or copy

### LINKED ISSUE #2771: Copy / move  from one subfolder to another subfolder does not work
### Steps to reproduce
1. setup:
- Create a root folder named folder_r_a with read permission for the user John
- Create a subfolder in folder_r_a with the name sub_a with write permission for the user John
- Create a root folder named folder_r_b with read permission for the user John
- Create a subfolder in folder_r_b with the name sub_b with write permission for the user John

2. the user John creates a file in sub_a

3. the user John selects the file and presses "copy or move"

4. in the new window, the user clicks on the home button to go to the root

5. then the folders "folder_r_a" and "folder_r_b" in the root directory are not visible in the window
-> This means that the user John cannot copy or move a file from sub_a to sub_b.

Only if John has write permissions for the folder_r_b, the folder name is displayed in the "Copy/Move - Window" and then it would possible to copy or move. But the write premission for the root folders would not be an option.

Use case:
There are several root directories that no user is allowed to write to. There are subdirectories in these directories where different users have different rights.
Users should now copy or move files from a subdirectory of a root directory to the subdirectory of another root directory.
For example, copy from administration to general information:
-Department
   |_ Administration
   |_ Finance
-Public
   |_ General information
   |_  ...

### Server configuration
**Nextcloud version:** (see Nextcloud admin page)
28.0.1
**Group folders version:**
16.0.3
**Updated from an older Nextcloud/ownCloud or fresh install:**
no
**Are you using external storage, if yes which one:** local/s3/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/Saml/...
no
### Client configuration
**Browser:**
various
**Operating system:**
various

### Logs
No log because no error or others related to this issue in log

### LINKED ISSUE #2926: NCv28: Users unable to move/copy into some group folders (OK in v27)
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Summary
NextCloud has been updated from version 27 to 28. The configuration of group folders with restricted permissions at different levels causes inconsistencies in the user interface, particularly when attempting to copy files into specific subfolders.

### Steps to reproduce
1. Configure group folders with restricted permissions:
   * Top-level folder (`level_one`) prohibits users from writing.
   * Subfolder (`level_two`) within level_one allows users to write and read.
2. Access the user web interface and attempt to copy files into the `level_two` subfolder. Note that the `level_two` subfolder may not appear in the copy/move window, despite being visible in the interface.

### Expected behaviour
Users should be able to see all relevant subfolders, including the `level_two` subfolder, when attempting to copy files within the group folder structure. The copy/move window should display all subfolders with appropriate permissions, allowing users to select and copy files as needed. (We had this in the previous NC version)

### Actual behaviour
In the current setup, the `level_two` subfolder may not appear in the copy/move window, despite being visible in the user interface. This discrepancy limits users' ability to copy files into specific subfolders, leading to frustration and inefficiency in file management tasks.

### Server configuration

**Operating system:**
Docker environment based on Alpine

**Web server:**
Nginx

**Database:**
mariadb:10.11

**PHP version:**
php-fpm 8.2.16

**Nextcloud version:** (see Nextcloud admin page)
28.0.3

**Group folders version:**
16.0.6

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from version 27

**Where did you install Nextcloud from:**

**Are you using external storage, if yes which one:** local/s3/smb/sftp/...
local

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/Saml/...
SAML

### Client configuration
**Browser:**
Chrome, Edge, Safary

**Operating system:**
Windows 10/11, MacOS

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

### LINKED ISSUE #44649: [Bug]: Copy/move to writable subfolder of write protected folder is not possible
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I write protect a folder (via low level file system permissions or with the ACLs of group folder app) and have a writeable folder inside it, this folder is not selectable on copy/move dialog on the web frontend.

I can copy files to that writeable subfolder if it is directly shared to me or if I use third party webdav access but not with the NC web interface. The issue is the same for regular folders or folders shared by someone else.

We need a fixed folder structure shared to our users and allow users to only write to deeper folders. Currently, they can upload and store files there but not move anything from their storage to the shared folder from NC UI.

### Steps to reproduce

1. create directory "folder" and "subfolder" inside it
2. go to low level files directory and issue "setfacl -m u:apache:rx folder ; setfacl -m u:apache:rwx folder/subfolder" (if PHP runs as apache)
3. rescan files with "occ files:scan"
4. move a arbitrary file or folder to "folder/subfolder", you cannot step into "folder" at all.

### Expected behavior

Folder should be visible, perhaps decorated with a write protection indicator.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by artonge**
With ACL, a groupfolders might be non-writable but still contain writable sub folders. 

- [x] Disable the submit button when we are in a non-writable folder @susnux or @skjnldsv, could one of you look into that?

Fix https://github.com/nextcloud/server/issues/44649
Fix https://github.com/nextcloud/groupfolders/issues/2771
Fix https://github.com/nextcloud/groupfolders/issues/2926

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -108,3 +108,7 @@
 
 AddDefaultCharset utf-8
 Options -Indexes
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
+
+ErrorDocument 403 /index.php/error/403
```
Comment (susnux) [inline .htaccess:113]: I do not think that this should be commited

**Comment by susnux** [comment]
> [ ] Disable the submit button when we are in a non-writable folder @susnux or @skjnldsv, could one of you look into that?

Doable with a small library change, I can do so.

**Comment by susnux** [comment]
> Doable with a small library change, I can do so.

@artonge we need the library change, but you can already change it in this PR:
Simply in the button facatory (next lines) add to each button:
`disabled: _selection.some((node) => node.permissions & Permission.CREATE) === 0)`

(better also rename `_selection` to remove the `_`)

**Comment by artonge** [comment]
> Simply in the button facatory (next lines) add to each button:
> `disabled: _selection.some((node) => node.permissions & Permission.CREATE) === 0)`

Don't we want to check the current folder's permissions instead of the selected nodes' permissions?

**Comment by susnux** [comment]
> Don't we want to check the current folder's permissions instead of the selected nodes' permissions?

We want to ensure the *picked* folder has permissions, so checking the selected node is the way to go :)
This already works (but will probably be false positive until we update the dialogs library)

**Review by susnux — APPROVED** [review]

**Comment by artonge** [comment]
- [ ] Needs a release with https://github.com/nextcloud/server/issues/45654 for tests to pass

**Review by skjnldsv — APPROVED** [review]

**Comment by susnux** [comment]
Rebased with the required library changes.

**Comment by artonge** [comment]
/backport! a0f3469323b9dc976d217c6f84cff410eb392cc7 to stable29

**Comment by artonge** [comment]
/backport! a0f3469323b9dc976d217c6f84cff410eb392cc7 to stable28

**Comment by artonge** [comment]
/backport! a0f3469323b9dc976d217c6f84cff410eb392cc7 to master

**Comment by blizzz** [comment]
rebased and commit results of `npm ci && npm run build &&  npm run postbuild` :four_leaf_clover:

**Comment by artonge** [comment]
PR split to move faster:
- https://github.com/nextcloud/server/pull/45981
- https://github.com/nextcloud/server/pull/45982

**Comment by artonge** [comment]
/backport ae0106ef9e9b4a91d385fff7490e17963e73c40d to stable29

**Comment by artonge** [comment]
/backport ae0106ef9e9b4a91d385fff7490e17963e73c40d to stable28

**Pull request merged**